cou/utils/juju_utils.py

```diff
@@ -201,6 +201,22 @@ def __init__(self, name: Optional[str]):
         self._juju_data = FileJujuData()
         self._model = Model(max_frame_size=JUJU_MAX_FRAME_SIZE, jujudata=self.juju_data)
         self._name = name
+        self._applications: dict[str, COUApplication] | None = None
+
+    @property
+    def applications(self) -> dict[str, COUApplication]:
+        """Return cached applications.
+
+        This property represents the cached output of the get_applications function.
+
+        :return: dictionary of COUApplication
+        :rtype: dict[str, COUApplication]
+        :raises ValueError: When get_applications has not yet been called.
+        """
+        if self._applications is None:
+            raise ValueError("The get_applications has not yet been called.")
```

Comment:
Why not call `get_applications` here instead of raising an error?

Reply:
async?

Reply:
Forgot about that :sweat_smile: